drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -367,6 +368,12 @@ int lis2dh_init(const struct device *dev)
 			    CONFIG_SENSOR_INIT_PRIORITY,		\
 			    &lis2dh_driver_api);
 
+#define IS_LSM303AGR_DEV(inst) \
+	DT_NODE_HAS_STATUS(DT_INST(inst, st_lsm303agr_accel), okay)
```

Comment:
Did you mean to use `DT_NODE_HAS_COMPAT` instead?

Reply:
Well, actually I'm not sure which is the best way. Here I need to understand whether a given instance for this driver is a LSM303AGR_ACCEL device or one of the others supported by the same driver. That's because there are some slight differences that should be handled.

If my undesratnd is correct DT_NODE_HAS_COMPAT is returning true if we have a given string in the compatibility, right? If so it seems reasonable to use this macro because in case of 303agr accel I would have:
    compatible = "st,lis2dh", "st,lsm303agr-accel";

Is that correct?

Reply:
This point is not clear at all to me. Maybe I shouild take a look better to documentation

Reply:
WHat about:
DT_NODE_HAS_COMPAT(DT_DRV_INST(inst), st_lsm303agr_accel)

It works fine on my side, and it looks more clear than before.

Reply:
>  DT_NODE_HAS_COMPAT(DT_DRV_INST(inst), st_lsm303agr_accel)

Yep, that's exactly what I had in mind, thanks